### PR TITLE
Migrate mock cloud from iotile_analytics to python_iotile_cloud

### LINKED ISSUE #18: Migrate mock_cloud from iotile_analytics here
This is to allow for wider testing of cloud api related python functionality.

Related Issue: https://github.com/iotile/iotile_analytics/issues/24

### PR DESCRIPTION
**Pull request by timburke**
The mock cloud spins up a local http server and answers api requests
in a compatible manner to the real iotile cloud.  Two pytest fixtures
are registered for use in any package that depends on
python_iotile_cloud:

- mock_cloud
- mock_cloud_nossl

In order to use the first mock_cloud that serves over SSL (with a
self-signed certificate), you must also have pyOpenSSL installed.
See: requirements-test.txt

Closes #18

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+### v0.7.4 (2018-01-26)
```
Comment (dkarchmer) [inline CHANGELOG.md]: Probably big enough to go to v0.8.0

File: setup.py [hunk 2]
```diff
@@ -14,6 +14,9 @@
         'iotile_cloud.utils',
         'iotile_cloud.stream'
     ],
+    entry_points={
+        'pytest11': ['iotile-analytics = iotile_cloud.utils.mock_cloud']
```
Comment (dkarchmer) [inline setup.py]: Is this line correct? `iotile-analytics`?

File: version.py [hunk 3]
```diff
@@ -1 +1 @@
-version = '0.7.3'
+version = '0.7.4'
```
Comment (dkarchmer) [inline version.py]: v0.8.0

**Review by dkarchmer — APPROVED** [review]
Let's call this v0.8.0. Thanks for doing the work.

File: setup.py (hunk 2 above)
Comment (timburke) [reply setup.py]: sorry, shouldn't affect anything but yes an oversight.

File: version.py (hunk 3 above)
Comment (timburke) [reply version.py]: understood.

**Comment by timburke** [comment]
Also, I want to hold on releasing for a bit while I add in the features from https://github.com/iotile/python_iotile_cloud/issues/20 to this PR since the first place we use it is going to require them.

**Pull request merged**